cmd/object-handlers.go

```diff
@@ -1618,6 +1655,27 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 	// Ensure that metadata does not contain sensitive information
 	crypto.RemoveSensitiveEntries(metadata)
 
+	var (
+		hasLifecycleConfig bool
+		goi                ObjectInfo
+		goiOpts            ObjectOptions
+		gerr               error
+	)
+	if _, err := globalBucketMetadataSys.GetLifecycleConfig(bucket); err == nil {
+		hasLifecycleConfig = true
+	}
+	if hasLifecycleConfig {
+		// Get appropriate object info to identify the remote object to delete
+		goiOpts = ObjectOptions{
+			VersionID: opts.VersionID,
+		}
```

Comment:
```suggestion
		goiOpts = ObjectOptions{}
```
versionID is generated by MinIO, not specified in request headers unless it is replication.